lightning/src/ln/onion_utils.rs

```diff
@@ -616,10 +706,11 @@ pub(crate) fn decode_next_hop(shared_secret: [u8; 32], hop_data: &[u8], hmac_byt
 					chacha_stream.read_exact(&mut next_bytes).unwrap();
 					assert_ne!(next_bytes[..], [0; 32][..]);
 				}
-				return Ok(Hop::Receive(msg));
+				return Ok((msg, None));
```

Comment:
I added comments to these lines indicating forward/receive. I kinda like it simple because if we add this, we'll either have more a bit more verbose callsites or need a wrapper, but don't feel strongly.